ft/ft/commands/filter.py

```diff
@@ -18,6 +17,9 @@ def handle_input(self, value):
         result = self.function(val_to_test)
 
         if result.fttype == T_BOOL:
+            if hasattr(self, 'negate_predicate'):
```

Comment:
Do we need this check? With your change, `negate_predicate` will be an attribute of `Filter`.

Reply:
To pass the tests, it seems necessary. If I remove it again, I get errors like this (by running `nosetests ft/tests`):

```
AttributeError: 'TakeWhile' object has no attribute 'negate_predicate'
```

Reply:
Might be due to the mock nature of the tests?

Reply:
Yes, this is due to the mock-nature of the tests. However, I think a cleaner way to solve this would be to set `self.negate_predicate = False` in the constructor.

Reply:
Of course, that makes sense. Thanks!